spec.html

```diff
@@ -30651,11 +30713,28 @@ <h1>String.prototype.valueOf ( )</h1>
 
       <emu-clause id="sec-string.prototype-@@iterator">
         <h1>String.prototype [ @@iterator ] ( )</h1>
-        <p>When the `@@iterator` method is called it returns an Iterator object (<emu-xref href="#sec-iterator-interface"></emu-xref>) that iterates over the code points of a String value, returning each code point as a String value. The following steps are taken:</p>
+        <p>When the `@@iterator` method is called it returns an Iterator object (<emu-xref href="#sec-iterator-interface"></emu-xref>) that iterates over the code points of a String value, returning each code point as a String value.</p>
+        <p>%String.prototype%[ @@iterator ] is a built-in generator function which </p>
+        <ul>
+          <li>the return iterator has the [[IteratorKind]] internal slot be "String Iterator".</li>
+          <li>the return iterator has the [[Prototype]] internal slot be %StringIteratorPrototype%.</li>
```

Comment:
these seem like they'd be slots on the function, not of the returned iterator, since they determine what the returned iterator is?

Reply:
the slot should on the iterator function, therefore the iterator can have a brand check to keep the current behavior.

```js
x = "abc"[Symbol.iterator]() // [[IteratorKind]] = "String Iterator"
function *y() {}
z = y() // [[IteratorKind]] not exists 
z.next.call(x) // brand check [[IteratorKind]] failed
```